word_cloud_generator/word_cloud_generator.py

```diff
@@ -0,0 +1,254 @@
+    )
+    parser.add_argument(
+        "-a",
+        "--assigned",
+        metavar="ASSIGNED",
+        help="Assigned user to get tickets from",
+    )
+    parser.add_argument(
+        "-f",
+        "--filter_for",
+        metavar="FILTER_FOR",
+        help="Strings to filter the word cloud for",
+    )
+    parser.add_argument(
+        "-n",
+        "--filter_not",
+        metavar="FILTER_NOT",
+        help="Strings to filter the word cloud to not have",
+    )
+    args = parser.parse_args(inp_args)
+    return args
+
+
+def get_response_json(auth, headers, url):
+    """
+    Function to send a get request to a url and return the response as json
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param url: The URL to send the request (string)
+    :returns: A dictionary of JSON values
+    """
+    session = requests.session()
+    session.headers = headers
+    session.auth = auth
+
+    attempts = 5
+
+    while attempts > 0:
+        response = session.get(url, timeout=5)
+        if (
+            response.content != b'{"status":"RUNNING"}'
+            and response.content != b'{"status":"ENQUEUED"}'
+        ):
+            break
+        else:
+            sleep(1)
+            attempts = attempts - 1
+
+    if attempts == 0:
+        raise requests.exceptions.Timeout(
+            "Get request status not completed before timeout"
+        )
+
+    return json.loads(response.text)
+
+
+def get_issues_contents_after_time(auth, headers, host, issue_filter):
+    """
+    Function to get the contents of through issues using a loop, as only 50 can be checked at a time
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param host: The host used to create the URL to send the request (string)
+    :returns: A list with the contents of all valid issues
+    """
+    issues_contents = []
+    return_amount = 50
+    issues_amount = 0
+    while return_amount == 50:
+        url = f"{host}/rest/servicedeskapi/servicedesk/6/queue/182/issue?start={issues_amount}"
+
+        json_load = get_response_json(auth, headers, url)
+
+        issues = json_load.get("values")
+
+        for issue in issues:
+            issue_date = datetime.strptime(
+                issue.get("fields").get("created")[:10], "%Y-%m-%d"
+            )
+            if issue_date < datetime.strptime(issue_filter.get("end_date"), "%Y-%m-%d"):
+                return issues_contents
+            if filter_issue(issue, issue_filter, issue_date):
+                issue_contents = issue.get("fields").get("summary")
+                if issue_contents:
+                    issues_contents.append(issue_contents)
+
+        return_amount = json_load.get("size")
+        issues_amount = issues_amount + return_amount
+
+    return issues_contents
```

Comment:
While the above comment is about changing the while loop - I personally don't like having a while loop check a condition - I prefer to use `while True` and use `break` to explicitly exit a loop on a condition. 
This is just easier for people to understand - you continue the loop until some condition which causes the loop to break

```suggestion
    curr_marker = 0
    check_limit = 50
    while True:
        url = f"{host}/rest/servicedeskapi/servicedesk/6/queue/182/issue?start={curr_marker}"
        json_load = get_response_json(auth, headers, url)
        issues = json_load.get("values")
        issues_contents = []
        i = 0
        for i, issue in enumerate(issues, 1):
            issue_date = datetime.strptime(
                issue.get("fields").get("created")[:10], "%Y-%m-%d"
            )

            if issue_date < datetime.strptime(issue_filter.get("end_date"), "%Y-%m-%d"):
                return issues_contents

            if filter_issue(issue, issue_filter):
                issue_contents = issue.get("fields").get("summary")
                if issue_contents:
                    issues_contents.append(issue_contents)

        if i < check_limit:
            break
        curr_marker += json_load.get("size")
    return issues_contents
```